mkosi/__init__.py

```diff
@@ -5945,12 +5947,13 @@ def parse_bytes(num_bytes: Optional[str]) -> Optional[int]:
     if factor > 1:
         num_bytes = num_bytes[:-1]
 
-    result = int(num_bytes) * factor
+    result = math.ceil(float(num_bytes) * factor)
     if result <= 0:
         raise ValueError("Size out of range")
 
-    if result % 512 != 0:
-        raise ValueError("Size not a multiple of 512")
+    rem = result % sector_size
+    if rem != 0:
+        result += sector_size - rem
```

Comment:
Maybe I'm misunderstanding, but taking the example of 7.3G, if you take the ceiling of 7.3 instead of the whole impression you get 8589934592 instead of 7838315316 before aligning to sector size.

Reply:
Ah, right.